src/lib/__tests__/search-client-test.js

```diff
@@ -0,0 +1,44 @@
+import InstantSearch from '../InstantSearch';
+
+describe('InstantSearch Search Client', () => {
+  it('gets called on search', () => {
+    const searchClientSpy = {
+      search: jest.fn(),
+    };
+
+    const search = new InstantSearch({
+      searchClient: searchClientSpy,
+    });
+
+    expect(searchClientSpy.search).not.toHaveBeenCalled();
+
+    search.start();
+
+    expect(search.helper.state.query).toBe('');
+    expect(searchClientSpy.search).toHaveBeenCalledTimes(1);
```

Comment:
also useful to assert with what it's been called?

Reply:
For your information this test was taken from: https://github.com/algolia/instantsearch.js/blob/develop/src/lib/__tests__/InstantSearch-test-2.js#L11

It's called with this:

```json
[
  {
    "indexName": "",
    "params": { "facets": [], "page": 0, "query": "test", "tagFilters": "" }
  }
]
```

Is it worth creating a snapshot for this?

Reply:
I'd go with a regular test here rather than a snapshot one, but not sure if it needs to be tested if already asserted enough in other spots

Reply:
Good call on testing the search() parameters here 👍  Considering the black box nature of the parameters, I'd go for a snapshot. Either would work (it's just more maintainance to not use snapshots)